src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs

```diff
@@ -4067,9 +4068,9 @@ public void WriteInvalidDepthPartial(bool formatted, bool skipValidation)
         [InlineData(true, false, ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")]
         [InlineData(false, true, ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")]
         [InlineData(false, false, ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")]
-        public void WriteComments(bool formatted, bool skipValidation, string comment)
+        public void WriteCommentsInArray(bool formatted, bool skipValidation, string comment)
```

Comment:
Really hard to tell just by looking at these tests, what format do we end up keeping in #75075?

Reply:
For #75075, because it was closed without decision about format, so I picked this format based on [comment](https://github.com/dotnet/runtime/issues/75075#issue-1361510964):
> I just guess current difference is not intentional. If we should make them same, which one should we pick ?
> 
> I updated code to make them same as this one, please decide if another is better.
> 
> ```
> {
>   "property": 
>   /*comment*/"value"
> ```

Which is existing format of string parameter overload of `WriteCommentValue()`, and because it is no matter which type of property value, so this format should be more align with #52408 

Note this PR is to fix both #75075 and #52408, I just updated PR description now.

Reply:
Hmm, on closer inspection both formats seem kind of strange to me, especially if the comment value contains line breaks. Could this possibly be
```json
{
    "property":
    /*comment*/
    "value"
}
```
But that's just me. Do we have any examples of other (non .NET) JSON formatters that handle comments?

@krwq @layomia opinions?

Reply:
I see that this differs depending on whether the property value is a string or object. What is the exact rule being used here?

Reply:
> I see from the tests that this differs depending on whether the property value is a string or object. What is the exact rule being used here?

The differ you found is just because current PR is to fix #52408. Current used rule is also based on those proposals, which is:

- for object and array: insert newlines between comment and object (or array), which is exactly described by #52408
- for other token types (not only string): keep existing behaviors not break, because #52408 doesn't  proposal them.

#75075 is to align 2 WriteCommentValue overload's different behavior about whether add newline between property name and comment, or not add newline. It is not related to property value currently because overload's difference is not there.

Current PR code should have been implemented based on rule above.

Reply:
@eiriktsarpalis what you proposed makes sense to me but also:

```json
{
    "property": /*comment*/
    "value"
}
```

would look good to me, I'm not strongly opinionated here - it seems like a weird place to put comment to me anyway